ntripping-cli/src/main.rs

```diff
@@ -0,0 +1,183 @@
+    #[clap(long)]
+    area_id: Option<u32>,
+
+    /// Field specifying which types of corrections are to be received
+    #[clap(long)]
+    corrections_mask: Option<u16>,
+
+    /// Solution ID, the identifier of the connection stream to reconnect to in the event of disconnections
+    #[clap(long)]
+    solution_id: Option<u8>,
+}
+
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error + Send + Sync>>;
+
+async fn run(opt: Cli) -> Result<()> {
+    let now = || -> DateTime<Utc> {
+        if let Some(epoch) = opt.epoch {
+            SystemTime::UNIX_EPOCH + Duration::from_secs(epoch.into())
+        } else {
+            SystemTime::now()
+        }
+        .into()
+    };
+
+    let auth = if let (Some(username), Some(password)) = (opt.username, opt.password) {
+        Some(Auth::new(username, password))
+    } else {
+        None
+    };
+
+    let msg: Sentence = if let Some(area_id) = opt.area_id {
+        Cra::new()
+            .with_area_id(area_id)
+            .with_corrections_mask(opt.corrections_mask)
+            .with_solution_id(opt.solution_id)
+            .into()
+    } else {
+        Gga::new()
+            .with_time(now())
+            .with_lat(opt.lat)
+            .with_lon(opt.lon)
+            .with_height(opt.height)
+            .into()
+    };
+
+    let client = {
+        let client = Client::new().with_client_id(opt.client_id).with_auth(auth);
+        if opt.gga_header {
+            client.with_ntrip_gga(msg)
+        } else {
+            client
+        }
+    };
+
+    let (mut sink, mut stream) = {
+        let url = if !opt.url.starts_with("http://") && !opt.url.starts_with("https://") {
+            format!("https://{}", opt.url)
+        } else {
+            opt.url
+        };
+        let uri = url.parse()?;
+        client.connect(uri).await?.split()
+    };
+
+    let writer_task = tokio::spawn(async move {
+        let mut out = io::stdout();
+        while let Some(data) = stream.try_next().await? {
+            out.write_all(&data).await?;
+        }
+        Result::Ok(())
+    });
+
+    if opt.gga_period == 0 {
+        return writer_task.await?;
+    }
+
+    let mut request_counter = opt.request_counter;
+    let mut gga_interval = tokio::time::interval(Duration::from_secs(opt.gga_period));
+    loop {
+        if writer_task.is_finished() {
+            break;
+        }
+        gga_interval.tick().await;
+        let sentence = msg.with_time(now()).with_request_counter(request_counter);
+        let _ = sink.send(sentence).await;
+        request_counter = request_counter.wrapping_add(1);
+    }
+
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    let opt = Cli::parse();
+    tracing_subscriber::fmt::fmt()
+        .with_max_level(match opt.verbose {
+            0 => LevelFilter::WARN,
+            1 => LevelFilter::DEBUG,
+            _ => LevelFilter::TRACE,
+        })
+        .compact()
+        .init();
+    tokio::runtime::Builder::new_current_thread()
```

Comment:
Default behavior has changed, with no user provided switches we now attempt to connect to Skylark and return error 403.  We should return the old behavior of doing nothing, or emit a help message if they don't specify anything.

Reply:
The current default behavior is to try and connect as well 

```
➜  ntripping git:(steve/updates) ✗ ntripping --verbose
*   Trying 50.112.176.233:2101...
* Connected to na.skylark.swiftnav.com (50.112.176.233) port 2101 (#0)
> GET /CRS HTTP/1.1
Host: na.skylark.swiftnav.com:2101
User-Agent: NTRIP ntrip-client/1.0
Accept: */*
Ntrip-Version: Ntrip/2.0
X-SwiftNav-Client-Id: 00000000-0000-0000-0000-000000000000
Expect: 100-continue

* Mark bundle as not supporting multiuse
< HTTP/1.1 401 Unauthorized
< Connection: close
< 
{ * Closing connection 0
➜  ntripping git:(steve/updates) ✗ echo $?
0
```

But right now this version returns a non-zero status code

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
     Running `target/debug/ntripping -v`
2022-12-15T22:56:06.736849Z DEBUG ntripping: connect addr=na.skylark.swiftnav.com:2101
2022-12-15T22:56:07.103324Z DEBUG hyper::proto::h1::io: flushed 215 bytes
2022-12-15T22:56:07.139357Z DEBUG hyper::proto::h1::io: parsed 1 headers
2022-12-15T22:56:07.139480Z DEBUG hyper::proto::h1::conn: incoming body is close-delimited
Error: BadStatus(401)
➜  ntripping git:(steve/updates) ✗ echo $?
1
```

But yeah the default behavior probably should exit successfully

Reply:
Is there a public endpoint we could default to? Maybe something that returns a source table?

Reply:
> The current default behavior is to try and connect as well

Hrmm, guess I never noticed it was trying to connect because it was returning success

Reply:
> Is there a public endpoint we could default to? Maybe something that returns a source table?

This returns a source table

![image](https://user-images.githubusercontent.com/183436/207989175-e226809b-4271-460d-8e69-6ecf9e0271df.png)

Reply:
Ah right, forgot that isn't protected by the auth